src/local_storage.rs

```diff
@@ -101,7 +104,7 @@ impl LocalStorage {
                     InternalError::InternalInvariantFailed
                 })
             })
-            .unwrap_or(Err(InternalError::StorageItemNotFound))
+            .map_or(Ok(None), |v| v.map(Some))
```

Comment:
This line changes a Option<Result> to a Result<Option>, though I'm not really sure if that's too clear

Reply:
Oh, hm. If we need to have a `Result` here either way then the `Option` is not actually any nicer. Sorry I didn't realize that when I wrote the issue. I would prefer the simpler return type (just plain `Result`) and have this also log an IIF with an error, like
```rust
.ok_or_else( || {
    error!("Tried to retrieve an item from storage that we never put there. T: {:?}, pid: {}",
        TypeId::of::<T>(),
        participant_id
    );
    InternalError::InternalInvariantFailed
)?
```

And maybe the same documentation note as in the non-mut version about calling `contains` first.

Reply:
Actually, I messed up here and didn't realize that the Result here is only there because an Option is being mapped to it. I can clean this up to just be an Option